internal/routing/aggregator.go

```diff
@@ -482,10 +482,21 @@ func (a *AggLogicalAndAggregator) Add(result interface{}, err error) error {
 		return e
 	}
 
+	// Atomic AND operation using CompareAndSwap loop (Go 1.21 compatible)
+	// TODO: Once minimum Go version is upgraded to 1.23+, replace this with:
+	//   if val { a.res.And(1) } else { a.res.And(0) }
+	var newVal int64
 	if val {
-		a.res.And(1)
+		newVal = 1
 	} else {
-		a.res.And(0)
+		newVal = 0
+	}
+	for {
+		old := a.res.Load()
+		desired := old & newVal
+		if a.res.CompareAndSwap(old, desired) {
+			break
+		}
 	}
```

Comment:
zero value of int64 is 0

```go
         var newVal int64 
         if val { 
                 newVal = 1 
         } 
         for { 
                 old := a.res.Load() 
                 desired := old & newVal 
                 if a.res.CompareAndSwap(old, desired) { 
                         break 
                 } 
          }
```

Reply:
But is there a point to use LoadCompare when val is false ?

Couldn't we return out the method?

Reply:
Yes. exactly